cmd/flags/admin.go

```diff
@@ -113,7 +113,8 @@ func (s *AdminServer) serveWithListener(l net.Listener) {
 	s.server = &http.Server{Handler: recoveryHandler(s.mux)}
 	s.logger.Info("Starting admin HTTP server", zap.Int("http-port", s.adminPort))
 	go func() {
-		if err := s.server.Serve(l); err != nil {
+		err := s.server.Serve(l)
+		if err != nil && err != http.ErrServerClosed {
```

Comment:
should we make it consistent with the other and use `case nil, http.ErrServerClosed, cmux.ErrListenerClosed:`?

Reply:
nice tip